### PR TITLE
[engine] Push choice authority through compilation pipeline

### PR DESCRIPTION
**Pull request by nickchapman-da**
Advance #15882

Push choice authority syntax though the LF compiler pipeline (haskell and scala sides) until we reach speedy.

### CODE REVIEW

File: daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala [hunk 1]
```diff
@@ -1637,7 +1637,7 @@ class DecodeV1Spec
       }
     }
 
-    "reject choice with observers if 1.7 < lf version < 1.dev" in {
+    "reject choice with observers if 1.7 < lf version < 1.dev" in { // TODO #15882 add similar test for choice authority version support
```
Comment (akrmn) [inline daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala:1640]: I think the upper bound here should be `1.11`, judging from `daml_lf_1.proto` https://github.com/digital-asset/daml/blob/c20fc05d78d21aaabdb556eb4abc4d5bdf59a776/daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf_1.proto#L43-L47

```suggestion
    "reject choice with observers if 1.7 < lf version < 1.11" in { // TODO #15882 add similar test for choice authority version support
```

**Review by akrmn — APPROVED** [review]
LGTM, thank you @nickchapman-da!

File: compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/EncodeV1.hs [hunk 2]
```diff
@@ -1002,12 +1002,18 @@ encodeChoiceObservers :: Maybe Expr -> Encode (Just P.Expr)
 encodeChoiceObservers chcObservers =
   encodeExpr (fromMaybe (ENil TParty) chcObservers)
 
+encodeChoiceAuthorizers :: Maybe Expr -> Encode (Just P.Expr)
+encodeChoiceAuthorizers = \case
+  Nothing -> pure Nothing -- dont add field to proto
```
[the remark this answers is not in the document]
Comment (nickchapman-da) [reply compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/EncodeV1.hs]: Thank you. I hadn't noticed this convention. Made suggested fix.

File: daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala (hunk 1 above)
Comment (nickchapman-da) [reply daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala:1640]: Thanks for the heads up. I will save this change for a future `1.dev-comment-fixup` PR.

**Pull request merged**